src/Rings/mpoly-localizations.jl

```diff
@@ -0,0 +1,751 @@
+    ambient_ring(S) == R || error("the multiplicative set is not contained in the given ring")
+    k = coefficient_ring(R)
+    R_loc = new{typeof(k), elem_type(k), RingType, elem_type(R), MultSetType}(R, S)
+    return R_loc
+  end
+end
+
+### required getter functions 
+base_ring(W::MPolyLocalizedRing) = W.R
+
+inverted_set(W::MPolyLocalizedRing) = W.S
+
+### required extension of the localization function
+localize_at(S::MPolyComplementOfPrimeIdeal) = MPolyLocalizedRing(ambient_ring(S), S)
+
+localize_at(S::MPolyComplementOfKPointIdeal) = MPolyLocalizedRing(ambient_ring(S), S)
+
+localize_at(S::MPolyPowersOfElement) = MPolyLocalizedRing(ambient_ring(S), S)
+
+### additional constructors
+MPolyLocalizedRing(R::RingType, P::MPolyIdeal{RingElemType}) where {RingType, RingElemType} = MPolyLocalizedRing(R, MPolyComplementOfPrimeIdeal(P))
+
+
+########################################################################
+# Elements of local polynomial rings                                   #
+########################################################################
+
+@Markdown.doc """
+MPolyLocalizedRingElem{
+    BaseRingType, 
+    BaseRingElemType,
+    RingType,
+    RingElemType, 
+    MultSetType
+  } <: AbsLocalizedRingElem{
+    RingType,
+    RingElemType, 
+    MultSetType
+  } 
+
+Elements of localizations of polynomial rings.
+"""
+mutable struct MPolyLocalizedRingElem{
+    BaseRingType, 
+    BaseRingElemType,
+    RingType,
+    RingElemType, 
+    MultSetType
+  } <: AbsLocalizedRingElem{
+    RingType,
+    RingElemType, 
+    MultSetType
+  } 
+
+  frac::AbstractAlgebra.Generic.Frac{RingElemType}
+  R_loc::MPolyLocalizedRing{BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType}
+
+  function MPolyLocalizedRingElem(
+      f::AbstractAlgebra.Generic.Frac{RingElemType}, 
+      R_loc::MPolyLocalizedRing{BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType}
+    ) where {BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType}
+
+    base_ring(parent(f)) == base_ring(R_loc) || error(
+	"the numerator and denominator of the given fraction do not belong to the original ring before localization"
+      )
+    denominator(f) in inverted_set(R_loc) || error(
+	"the given denominator is not admissible for this localization"
+      )
+    return new{BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType}(f, R_loc)
+  end
+end
+
+### required getter functions 
+numerator(a::MPolyLocalizedRingElem) = numerator(a.frac)
+
+denominator(a::MPolyLocalizedRingElem) = denominator(a.frac)
+
+parent(a::MPolyLocalizedRingElem) = a.R_loc
+
+### additional getter functions
+fraction(a::MPolyLocalizedRingElem) = a.frac
+
+### required conversions
+(W::MPolyLocalizedRing{BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType})(f::RingElemType) where {BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType} = MPolyLocalizedRingElem((FractionField(base_ring(W)))(f), W)
```

Comment:
Such super long lines are hard to read. I'd use at most ~100 chars per line and then wrap. E.g.
```suggestion
(W::MPolyLocalizedRing{BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType})(f::RingElemType)
 where {BaseRingType, BaseRingElemType, RingType, RingElemType, MultSetType}
   = MPolyLocalizedRingElem((FractionField(base_ring(W)))(f), W)
```

And/or abbreviate types you don't actually need; and omit any trailing ones that are unused. So:
```suggestion
(W::MPolyLocalizedRing{T1, T2, T3, RingElemType})(f::RingElemType) where {T1, T2, T3, RingElemType} = MPolyLocalizedRingElem((FractionField(base_ring(W)))(f), W)
```

Reply:
I totally agree on long lines being annoying. But I found it hard to do appropriate wrappings in Julia. You always have to find specific places which allow you to introduce a line break without breaking the code. For instance, you can't just start the `where` block in a new line, but you have to break between some two arguments in either one of the lists. That always had the air of a hack to me and I don't know how one is supposed to address this issue in general.

Reply:
Btw: I started abbreviating the types further down in the section on ideals. However, I find it very annoying, if the type's name doesn't reveal what they stand for.